libimage/load.go

```diff
@@ -28,66 +28,69 @@ func (r *Runtime) Load(ctx context.Context, path string, options *LoadOptions) (
+
+	for _, f := range []func() ([]string, string, error){
 		// OCI
-		func() ([]string, error) {
+		func() ([]string, string, error) {
 			logrus.Debugf("-> Attempting to load %q as an OCI directory", path)
 			ref, err := ociTransport.NewReference(path, "")
 			if err != nil {
-				return nil, err
+				return nil, ociTransport.Transport.Name(), err
 			}
-			return r.copyFromDefault(ctx, ref, &options.CopyOptions)
+			images, err := r.copyFromDefault(ctx, ref, &options.CopyOptions)
+			return images, ociTransport.Transport.Name(), err
 		},
 
 		// OCI-ARCHIVE
-		func() ([]string, error) {
+		func() ([]string, string, error) {
 			logrus.Debugf("-> Attempting to load %q as an OCI archive", path)
 			ref, err := ociArchiveTransport.NewReference(path, "")
 			if err != nil {
-				return nil, err
+				return nil, ociArchiveTransport.Transport.Name(), err
 			}
-			return r.copyFromDefault(ctx, ref, &options.CopyOptions)
+			images, err := r.copyFromDefault(ctx, ref, &options.CopyOptions)
+			return images, ociArchiveTransport.Transport.Name(), err
 		},
 
 		// DIR
-		func() ([]string, error) {
+		func() ([]string, string, error) {
 			logrus.Debugf("-> Attempting to load %q as a Docker dir", path)
 			ref, err := dirTransport.NewReference(path)
 			if err != nil {
-				return nil, err
+				return nil, dirTransport.Transport.Name(), err
 			}
-			return r.copyFromDefault(ctx, ref, &options.CopyOptions)
+			images, err := r.copyFromDefault(ctx, ref, &options.CopyOptions)
+			return images, dirTransport.Transport.Name(), err
 		},
 
 		// DOCKER-ARCHIVE
-		func() ([]string, error) {
+		func() ([]string, string, error) {
 			logrus.Debugf("-> Attempting to load %q as a Docker archive", path)
 			ref, err := dockerArchiveTransport.ParseReference(path)
 			if err != nil {
-				return nil, err
+				return nil, dockerArchiveTransport.Transport.Name(), err
 			}
-			return r.loadMultiImageDockerArchive(ctx, ref, &options.CopyOptions)
-		},
-
-		// Give a decent error message if nothing above worked.
-		func() ([]string, error) {
-			return nil, errors.New("payload does not match any of the supported image formats (oci, oci-archive, dir, docker-archive)")
+			images, err := r.loadMultiImageDockerArchive(ctx, ref, &options.CopyOptions)
+			return images, dockerArchiveTransport.Transport.Name(), err
 		},
 	} {
-		loadedImages, loadError = f()
-		if loadError == nil {
-			return loadedImages, loadError
+		loadedImages, transportName, err := f()
+		if err == nil {
+			return loadedImages, nil
 		}
-		logrus.Debugf("Error loading %s: %v", path, loadError)
+		logrus.Debugf("Error loading %s (%s): %v", path, transportName, err)
+		loadErrors = append(loadErrors, fmt.Errorf("%s: %v", transportName, err))
+	}
+
+	// Give a decent error message if nothing above worked.
+	loadError := fmt.Errorf("payload does not match any of the supported image formats:")
+	for _, err := range loadErrors {
+		loadError = fmt.Errorf("%w\n * %v", loadError, err)
```

Comment:
Why do you wrap the loadError? It is impossible to match anyway since it is not exported as constant or type.

Reply:
That really makes no sense, changed